include/material/mohr_coulomb.tcc

```diff
@@ -471,49 +469,49 @@ Eigen::Matrix<double, 6, 1> mpm::MohrCoulomb<Tdim>::compute_stress(
     (*state_vars)["cohesion"] = cohesion_residual_;
   }
 
-  // Yield function for the current stress state
-  Eigen::Matrix<double, 2, 1> yield_function;
-  auto yield_type = this->compute_yield_state(&yield_function, state_vars);
-  Vector6d df_dsigma = Vector6d::Zero();
-  Vector6d dp_dsigma = Vector6d::Zero();
-  double softening = 0.;
-
-  // compute based on the stress after checked
-  this->compute_stress_invariants(stress, state_vars);
-  yield_type = this->compute_yield_state(&yield_function, state_vars);
-  this->compute_df_dp(yield_type, state_vars, stress, &df_dsigma, &dp_dsigma,
-                      &softening);
-
-  double lambda = df_dsigma.dot(this->de_ * dstrain) /
-                  ((df_dsigma.dot(this->de_ * dp_dsigma)) + softening);
-
-  // compute the trial stress
+  // Compute the trial stress
   Vector6d trial_stress = stress + (this->de_ * dstrain);
+  // Compute stress invariants based on trial stress
   this->compute_stress_invariants(trial_stress, state_vars);
-
-  // Trial yield function
+  // Compute yield function based on the trial stress
   Eigen::Matrix<double, 2, 1> yield_function_trial;
   auto yield_type_trial =
       this->compute_yield_state(&yield_function_trial, state_vars);
 
+  // Return the updated stress in elastic state
+  if (yield_type_trial == FailureState::Elastic) return trial_stress;
+
+  // Correct the stress back to the yield surface if it is in plastic state
+  // Compute plastic multiplier based on trial stress (Lambda trial)
   double softening_trial = 0.;
   Vector6d df_dsigma_trial = Vector6d::Zero();
   Vector6d dp_dsigma_trial = Vector6d::Zero();
   this->compute_df_dp(yield_type_trial, state_vars, trial_stress,
                       &df_dsigma_trial, &dp_dsigma_trial, &softening_trial);
-
-  // Lambda trial
   double yield = 0.;
   if (yield_type_trial == FailureState::Tensile)
     yield = yield_function_trial(0);
   if (yield_type_trial == FailureState::Shear) yield = yield_function_trial(1);
-
   double lambda_trial =
       yield /
       ((df_dsigma_trial.transpose() * de_).dot(dp_dsigma_trial.transpose()) +
        softening_trial);
 
-  // compute the correction stress
+  // Compute stress invariants based on stress input
+  this->compute_stress_invariants(stress, state_vars);
+  // Compute yield function based on stress input
+  Eigen::Matrix<double, 2, 1> yield_function;
+  auto yield_type = this->compute_yield_state(&yield_function, state_vars);
+  // Compute plastic multiplier based on stress input (Lambda)
+  double softening = 0.;
+  Vector6d df_dsigma = Vector6d::Zero();
```

Comment:
Is this Eigen::Vector6d? Could you please use just `auto` as you are using `Vector6d::Zero()`

Reply:
If I use auto, it will be defined as  Eigen::Matrix<double, 6, 1> but not Vector6D. Although  Vector6D is just  Eigen::Matrix<double, 6, 1>, there will be treated as different types in the arguments of functions. If you want the "auto", I need to change all the arguments types of functions from Vector6D to Eigen::Matrix<double, 6, 1>.

Reply:
There will be an error happen when using auto, because it will set the value as "Eigen::Matrix<double, 6, 1>" but not "Vector6d". Although they are the same one, they are treated as two different types arguments in functions

Reply:
Could you please use `Eigen::Vector6d df_dsigma = Eigen::Vector6d::Zero()`